lib/fs.js

```diff
@@ -226,12 +225,13 @@ fs.existsSync = function(path) {
 fs.readFile = function(path, options, callback_) {
   var callback = maybeCallback(arguments[arguments.length - 1]);
 
-  if (!options || typeof options === 'function')
+  if (!options || typeof options === 'function') {
     options = { encoding: null, flag: 'r' };
-  else if (typeof options === 'string')
+  } else if (typeof options === 'string') {
     options = { encoding: options, flag: 'r' };
-  else if (typeof options !== 'object')
-    throw new TypeError('Bad arguments');
+  } else if (typeof options !== 'object') {
+    throwOptionsError(options);
+  }
```

Comment:
@thefourtheye it is probably this [eslint bug](https://github.com/eslint/eslint/issues/2343) again. We already encountered it in https://github.com/nodejs/io.js/issues/1742